### PR TITLE
Pinning gym to 0.21 to fix test issues

### PR DESCRIPTION
**Pull request by avnishn**
see https://github.com/openai/gym/pull/2639

for more details

tldr; Theres some issue with RandomNumberGenerator pickling and the pr even mentions RLlib as being broken by gym 0.22

This breakage is also causing issues with the red tests on flakey tests ray io:
image

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — APPROVED** [review]
thanks man!

**Review by sven1977 — APPROVED** [review]
